### ISSUE TITLE
Changing the spelling of datas to data

### ISSUE THREAD

**Issue by terop** [ttu/ruuvitag-sensor#157]
As the correct correct spelling for data is data and not data**s**  I would like to correct this. However as there is (at least) one function (`get_datas()`) with datas in the name doing this change would cause an API breakage.
How should I go about tho make this change? I see two options here:
1) change all datas to data which would cause the API breakage
or
2) only change variables and documentation but not touch functions containing datas in their name

@ttu which option would you prefer?

**Comment by ttu** [comment]
Hi! Thanks for bringing this up.

Breaking API is not often a good choice, especially in cases where we do not really make any significant improvement. But I agree that we should fix this.

My suggestion would be that we add get_data-method to RuuviTagSensor, but we do not remove get_datas-method, but it is marked as deprecated. I don't know what is recommended practice to mark it in Python, but we can just add a comment if Python doesn't have e.g. attribute for it.

ruuvi.py:
```py
@staticmethod
def get_data(callback, macs=[], run_flag=RunFlag(), bt_device=''):
  # Move implementation here

@staticmethod
def get_datas(callback, macs=[], run_flag=RunFlag(), bt_device=''):
  """
  DEPRECATED
  This method will be removed in future vetsion
  Use get_data-method instead
  """
  return RuuviTagSensor.get_data(callback, macs, run_flag, bt_device)
```

Changes:
1. Add get_data and mark get_datas as deprecated from RuuviTagSensor
1. Change all other variables and function names
1. Update documentations

Would you like to work on this? We could add these changes to package before we release version 2.0.

I'm currently adding get_datas_async-method (https://github.com/ttu/ruuvitag-sensor/pull/154), but will rename it to get_data_async 😄

**Comment by ukBaz** [comment]
You might also want to add a [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning) to the `get_datas` method

**Comment by ttu** [comment]
Noticed that RuuviTagSensor already has a get_data-method. It is not "public" and there is no mention of it in README. It is only used in RuuviTag-class and even in the README there is a mention that it should not be used. Originally there was a plan that it would have been used with some home automation system.

The method returns first raw data from the selected sensor, so we can rename it to get_first_raw_data.

Adapters also have a get_data and get_datas-methods. get_data returns first broadcasted data, so that function can be renamed to get_first_data and get_datas to get_data.

**Comment by ttu** [comment]
@ukBaz Thanks! Wasn't aware of FutureWarning-exception.

We do not have any urgency to remove `get_datas`-method, so maybe we can add that exception in some future version?

**Comment by terop** [comment]
Sorry for the delayed response due to me being abroad without the possibility to work on this.

I was and still am planning to work on this. I raised this issue to see if the change would be desirable and possibly how it should be done. I have already started work on option 2) from my original message and will do the changes suggested in Tomi's first comment. I will rebase my work on top of the changes in PR #159 so merging it sooner rather than later would be good.

As for the FutureWarning I recommend that it is added in conjunction in fixing this issue. The rationale for this is that probably not too many developers read the code and having them seeing the warning earlier rather than later would be good so that existing code can be fixed. The change is not very big but it is anyway good to highlight the need to change the code.

**Comment by ttu** [comment]
PR #159 is now merged. https://github.com/terop/ruuvitag-sensor/tree/datas_to_data looks good 👍 I will test it with RuuviTags when it is ready to be merged.

I wasn't familiar with how warn and FutureWarning works, but that looks extremely good. You are correct, let's add it immediately.

**Comment by terop** [comment]
PR opened and is available at https://github.com/ttu/ruuvitag-sensor/pull/164.

**Comment by ttu** [comment]
Method and variable renames were released with version `2.0`
https://github.com/ttu/ruuvitag-sensor/blob/master/CHANGELOG.md#200---2022-08-03
